pwndbg/commands/setrlimit.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import argparse
+
+import pwndbg.color.message as message
+import pwndbg.commands
+from pwndbg.commands import CommandCategory
+
+if pwndbg.dbg.is_gdblib_available():
+    import gdb
+
+RLIM_INFINITY = -1
+LIMITS: dict[str, int] = {
+    "core": 4,
+    "cpu": 0,
+    "fsize": 1,
+    "data": 2,
+    "stack": 3,
+    "nofile": 7,
+    "as": 9,
+}
+
+parser = argparse.ArgumentParser(
+    description="Set a POSIX resource limit in the debugged process.",
+)
+parser.add_argument(
+    "resource",
+    type=str,
+    help="Which resource to limit: " + ", ".join(LIMITS.keys()),
+)
+parser.add_argument(
+    "soft",
+    type=str,
+    help="Soft limit value (integer) or 'infinite'",
+)
+parser.add_argument(
+    "hard",
+    type=str,
+    nargs="?",
+    default=None,
+    help="Hard limit value (integer) or 'infinite' (defaults to soft)",
+)
+
+
+def _invoke_setrlimit(num: int, soft_val: int, hard_val: int) -> str:
+    return gdb.execute(
+        f"call (int) setrlimit({num}, (struct rlimit[]){{ {{ {soft_val}, {hard_val} }} }})",
```

Comment:
Ideally, we should have a fallback which would use shellcode execution for this iff the function is not available etc.